core/src/test/kotlin/xyz/block/artifactswap/core/utils/RepoNameProviderTest.kt

```diff
@@ -0,0 +1,78 @@
+package xyz.block.artifactswap.core.utils
+
+import java.nio.file.Path
+import kotlin.io.path.createDirectories
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.transport.URIish
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+
+class RepoNameProviderTest {
+
+  @TempDir lateinit var tempDir: Path
+
+  @Test
+  fun `repo name is correct when root dir is git repo`() {
+    val expectedRepoName = "test-repo"
+    val repoRoot =
+      tempDir.resolve(expectedRepoName).also { repoRoot ->
+        Git.init().setDirectory(repoRoot.toFile()).call().use { git ->
+          git
+            .remoteAdd()
+            .setName("origin")
+            .setUri(URIish("https://example.com/$expectedRepoName.git"))
```

Comment:
for a more realistic test I'd fake out some github repos like 

* `https://github.com/block/artifact-swap.git`
* `org-123123123@github.com:block/artifact-swap.git` (gh enterprise style)

your parsing should work for both though